improver/wxcode/utilities.py

```diff
@@ -102,22 +102,49 @@
 }
 
 
-def update_tree_units(tree: Dict[str, Dict[str, Any]]) -> Dict[str, Dict[str, Any]]:
+def update_tree_thresholds(
+    tree: Dict[str, Dict[str, Any]], target_period: Optional[int] = None
+) -> Dict[str, Dict[str, Any]]:
     """
     Replaces value / unit pairs from tree definition with an Iris AuxCoord
-    that encodes the same information.
+    that encodes the same information. Also scales any threshold values that
+    have an associated period (e.g. accumulation in 3600 seconds) by a factor
+    to reflect the target period (e.g. a 3-hour, 10800 second, weather symbol).
 
     Args:
         tree:
             Weather symbols decision tree.
+        target_period:
+            The period in seconds that the weather symbol being produced should
+            represent. This should correspond with any period diagnostics, e.g.
+            precipitation accumulation, being used as input. This is used to scale
+            any threshold values that are defined with an associated period in
+            the decision tree.
     Returns:
-        The tree now containing AuxCoords instead of value / unit pairs.
+        The tree now containing AuxCoords instead of value / unit pairs, with
+        period diagnostic threshold values scaled appropriately.
+
+    Raises:
+        ValueError: If thresholds are defined with an associated period and no
+                    target_period is provided.
     """
 
     def _make_thresholds_with_units(items):
         if isinstance(items[0], list):
             return [_make_thresholds_with_units(item) for item in items]
-        values, units = items
+        try:
+            values, units, period = items
```

Comment:
In the interests of the decision tree being self-describing, we should include the period units too. Doing so would allow us to state these values as `1, "hours"`, which is more instantly comprehendible than `3600`. I'm not too fussed about this though, so if you'd rather leave it as it is, I won't argue.

Reply:
I don't think we modify this enough to warrant the work, and the documentation covers this. As such I've taken the easy option and left it.